python/lsst/daf/butler/registry/dimensions/spatial.py

```diff
@@ -53,10 +56,18 @@ def __init__(self, db: Database, element: DimensionElement, *, table: sqlalchemy
         self._commonSkyPixOverlapTable = commonSkyPixOverlapTable
         assert element.spatial
 
-    def getCommonSkyPixOverlapTable(self, dataId: Optional[DataId] = None) -> sqlalchemy.schema.Table:
-        # Docstring inherited from
-        # DimensionRecordStorage.getCommonSkyPixOverlapTable.
-        return self._commonSkyPixOverlapTable
+    def join(
+        self,
+        builder: QueryBuilder, *,
+        regions: Optional[NamedKeyDict[DimensionElement, sqlalchemy.sql.ColumnElement]] = None,
+        timespans: Optional[NamedKeyDict[DimensionElement, Timespan[sqlalchemy.sql.ColumnElement]]] = None,
+    ):
```

Comment:
I may not be following along with code flow, but what if this function gets called before one in which regions are added, should what is being added here get added some how?

Reply:
The base class docs (looks like I need to add a comment pointing to them here) clarify that this method should only ever be called by `QueryBuilder.joinDimensionElement` (this is a visitor pattern thing), and it's responsible for guaranteeing that `regions` is provided (even if it is an empty dict, which is fine) only when appropriate.